### ISSUE TITLE
[bug] `switchLanguage` not working when targeting the original page language + scroll to top

### ISSUE THREAD

**Issue by WarningImHack3r** [inlang/inlang#1237]
### Did you check if the config file includes errors?

- [X] Yes, I have validated the config file

### Problem

It used to work fine in previous versions like `0.7.1`, but it is not working anymore as of the latest version(s).

I have a EN/FR toggle switch on my page. Let's say I first load my page in French because my system is in French, I can switch to English with that toggle, but I cannot switch back to French without needing a full page reload.

The second problem is that switching languages scrolls to the top of the page. It used to be a seamless in-place text change back in these older versions, and I miss this behavior.

### Expected behavior

The switch language to always work, and the language switch to not scroll to the top (when it works).

### Reproduction

Same repository as always: https://github.com/EmeraldHQ/Website. Check the Dependabot PR (https://github.com/EmeraldHQ/Website/pull/79) for the latest version with these issues, and the `main` branch for the good old behavior.

### System Info

```Shell
System:
    OS: macOS 13.5
    CPU: (10) arm64 Apple M1 Pro
    Memory: 90.20 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.5.1 - /opt/homebrew/bin/node
    npm: 9.8.1 - /opt/homebrew/bin/npm
    pnpm: 8.6.12 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 115.0.5790.170
    Safari: 16.6
    Safari Technology Preview: 17.0
  npmPackages:
    @inlang/sdk-js: ^0.7.1 => 0.7.1 
    @sveltejs/adapter-static: ^2.0.3 => 2.0.3 
    @sveltejs/kit: ^1.22.4 => 1.22.4 
    svelte: ^4.1.2 => 4.1.2
```

**Comment by ivanhofer** [comment]
Thanks for giving us so much feedback!

We haven't enabled full reactivity yet, but we plan to do so in the future. We currently have a workaround in place that reloads all data and was missing some implementation detail.

Those two issues should be fixed in version `0.11.8`.

**Comment by WarningImHack3r** [comment]
Good!

> We haven't enabled full reactivity yet

What do you mean? How did it work in earlier versions then?

**Comment by ivanhofer** [comment]
We had hired a contractor to build the initial version due to time pressure. But that implementation did not scale well and we rewrote everything from scratch. And there we decided to not offer full reactivity for now as it requires additional development effort. Initially thought that we can add it a few weeks later, but this still needs to wait until we finish the core refactoring of the inlang config structure.
